viewing_party/party.py

```diff
@@ -1,23 +1,188 @@
+    for movies in user_data["watched"]:
+        for values in movies:
+            if movies["genre"] not in most_watched_genre:
+                most_watched_genre[movies["genre"]] = 1
+            else:
+                most_watched_genre[movies["genre"]] += 1
+    length_of_movies = len(user_data["watched"])
+    if length_of_movies == 0:
+        return None
+    else:
+        most_watched_genre = max(
+            most_watched_genre, key=most_watched_genre.get)
+        return most_watched_genre
+
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+
+def get_unique_watched(user_data):
+    user_unique_watched = []
+    user_watched = []
+    friends_watched = []
+
+    for movies in user_data["watched"]:
+        user_watched.append(movies)
+
+    for movies in user_data["friends"]:
+        for details in movies["watched"]:
+            friends_watched.append(details)
+
+    user_unique_watched = [i for i in user_watched if i not in friends_watched]
+
+    if len(user_unique_watched) == []:
+        return 0
+    else:
+        return user_unique_watched
+
+
+def get_friends_unique_watched(user_data):
+    friends_unique_watched = []
+    user_watched = []
+    friends_watched = []
+
+    for movies in user_data["watched"]:
+        user_watched.append(movies)
+
+    for movies in user_data["friends"]:
+        for details in movies["watched"]:
+            friends_watched.append(details)
+
+    friends_unique_watched = [
+        i for i in friends_watched if i not in user_watched]
+
+    no_duplicates_movie_list = []
+    for i in friends_unique_watched:
+        if i not in no_duplicates_movie_list:
+            no_duplicates_movie_list.append(i)
+
+    if len(friends_unique_watched) == []:
+        return 0
+    else:
+        return no_duplicates_movie_list
+
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    recommend_list = []
+    combined_list = []
+    subscriptions = []
+    for movies in user_data["subscriptions"]:
+        subscriptions.append(movies)
+
+    friends_list = get_friends_unique_watched(user_data)
+    users_list = get_unique_watched(user_data)
+    combined_list = friends_list + users_list
+
+    for item in friends_list:
+        for details in item:
+            if item["host"] in subscriptions:
+                recommend_list.append(item)
+
+    newlist = []
+    for i in recommend_list:
+        if i not in newlist:
+            newlist.append(i)
+
+    if len(newlist) == []:
+        return 0
+    else:
+        return newlist
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+
+def get_new_rec_by_genre(user_data):
+    new_rec_genre = []
+    friends_list = get_friends_unique_watched(user_data)

```

Comment:
I think to save you some filtering we could use `get_avilable_recs` method instead!

```suggestion
    recs_list = get_available_recs(user_data)
```